### PR TITLE
Add anyMatch() and noneMatch() (to go along with allMatch())

### LINKED ISSUE #1126: Add noneMatch() and anyMatch() to AbstractListAssert
#### Summary
Working with code using streams, I saw code like this:

```java
assertThat(acceptedStrings.stream().allMatch(filter::matches)).isTrue();
assertThat(rejectedStrings.stream().noneMatch(filter::matches)).isTrue();
```

I thought about harnessing AssertJ power to improve the error messages, so I found out I could use the following:

```java
assertThat(acceptedStrings).allMatch(filter::matches);
```

However, I could not find an equivalent for `Stream.noneMatch()`, and noticed `Stream.anyMatch()` has no equivalent in `ObjectEnumerableAssert` either!.

It would be great if `ObjectEnumerableAssert.noneMatch()` and `ObjectEnumerableAssert.anyMatch()` could be added.

#### Java 8 specific ?

* YES : create Pull Request from the `master` branch

### PR DESCRIPTION
**Pull request by JnRouvignac**
#### Check List:
* Fixes #1126
* Unit tests : YES
* Javadoc with a code example (API only) : YES

### CODE REVIEW

**Comment by joel-costigliola** [comment]
I haven't fully reviewed but I notices a few things.
The build fails due to a compilation error as the added methods are not defined in `ObjectEnumerableAssert` (maybe you have simply forgotten to commit it).
Morever I don't see any javadoc in the API which should be different in `AbstractIterableAssert`, `AbstractObjectArrayAssert` and `AtomicReferenceArrayAssert`.

**Comment by JnRouvignac** [comment]
> The build fails due to a compilation error as the added methods are not defined in ObjectEnumerableAssert (maybe you have simply forgotten to commit it).

Apologies, I had it, but then I removed it with a wrong manipulation.

> Morever I don't see any javadoc in the API which should be different in AbstractIterableAssert, AbstractObjectArrayAssert and AtomicReferenceArrayAssert.

Right, I should have specialized javadocs for each of these.
I am going to fix this now.

Thanks for the quick review.

**Comment by JnRouvignac** [comment]
All the preliminary comments have been addressed.
Please review when you get a chance.

Thanks again for your extremely quick feedback.

**Comment by joel-costigliola** [comment]
Thanks @JnRouvignac for the quick update, I'll review it and integrate it by the end of the week if there are only minor changes to do.

**Comment by JnRouvignac** [comment]
The last push resolves conflicts with `master`.

**Comment by joel-costigliola** [comment]
The PR looks good but it lacks of unit tests for:

- `Iterables.assertAnyMatch` 
- `Iterables.assertNoneMatch`
- `AnyElementShouldMatch`
For the first two add some tests like `Iterables_assertAllMatch_Test` and for the last `ElementsShouldSatisfy_create_Test`.

File: src/main/java/org/assertj/core/error/AnyElementShouldMatch.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2017 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import org.assertj.core.presentation.PredicateDescription;
+
+public class AnyElementShouldMatch extends BasicErrorMessageFactory {
+  private static final String SINGLE_NON_MATCHING_ELEMENT = "%nExpecting any element of:%n  <%s>%nto match %s predicate but this element did not:%n  <%s>";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/AnyElementShouldMatch.java]: the error message is logically incorrect, it should be:
`"%nExpecting any element of:%n  <%s>%nto match %s predicate but non did."`

File: src/main/java/org/assertj/core/error/AnyElementShouldMatch.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  private static final String MULTIPLE_NON_MATCHING_ELEMENT = "%nExpecting any element of:%n  <%s>%nto match %s predicate but these elements did not:%n  <%s>";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/AnyElementShouldMatch.java]: Remove, this case is not relevant for anyMatch

**Comment by JnRouvignac** [comment]
> The PR looks good but it lacks of unit tests for:
> * `Iterables.assertAnyMatch`
> * `AnyElementShouldMatch`
>
> For the first add some tests like `Iterables_assertAllMatch_Test` and for the last `ElementsShouldSatisfy_create_Test`.

Thanks, I knew I was missing something but did not know where to add it.
It also helped me see the other problems you mentioned.

I pushed a new separate commit to fix all the reported issues to ease your review.
Please let me know if you want me to squash the commits before merging.

**Comment by joel-costigliola** [comment]
Integrated thanks @JnRouvignac !

**Comment by JnRouvignac** [comment]
Woohoo! Thanks a lot for your time!

**Pull request closed without merging**